distributed/stealing.py

```diff
@@ -134,8 +133,10 @@ def steal_time_ratio(self, ts):
         if cost_multiplier > 100:
             return None, None
 
-        level = int(round(log(cost_multiplier) / log_2 + 6, 0))
-        level = max(1, level)
+        level = int(round(log2(cost_multiplier) + 6))
+        if level < 1:
+            level = 1
```

Comment:
Using `if` is roughly an order of magnitude faster than `max`.

```python
In [1]: %timeit max(1, 3)
222 ns ± 1.66 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [2]: %timeit max(1, 0)
222 ns ± 1.41 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [3]: %%timeit
   ...: L = 3
   ...: if L < 1:
   ...:     L = 1
   ...: 
25.5 ns ± 0.165 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [4]: %%timeit
   ...: L = 0
   ...: if L < 1:
   ...:     L = 1
   ...: 
29.7 ns ± 0.231 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
No objection here, but I do want us to look at the full picture rather than just micro-optimizing.  200ns may not matter in the grand scheme.  At some point it might make more sense to keep code around that is nanoseconds slower because it is more readable.  

I'm not making a judgement in this case (I haven't looked at this much).   I just want to avoid us going overboard here.

Reply:
Mainly came here since `put_key_in_stealable` takes as much runtime as `transition_waiting_processing` and `transition_processing_memory`. With the latter 2 we have already put a lot of time into optimizing them, but this part of the code has largely been untouched. So am trying to tackle some of the low hanging fruit here.

Reply:
Should add that collectively these changes more than halve the time spent in `put_key_in_stealable`.

Reply:
Glad to hear it.  My comment was just general.  I have no particular concern with this PR.

Reply:
Understood. Just wanted to provide context :)